app/models/concerns/indexable.rb

```diff
@@ -101,30 +101,31 @@ def query(query, options={})
       options[:page][:number] ||= 1
       options[:page][:size] ||= 25
 
-      # enable cursor-based pagination for DOIs
-      if self.name == "Doi" && options.dig(:page, :cursor).present?
+      # Default sort should be whatever is passed in
+      sort = options[:sort]
+
+      # Cursor nav use the search after, this should always be an array of values that match the sort.
+      if options.dig(:page, :cursor).present?
         from = 0
         search_after = Array.wrap(options.dig(:page, :cursor))
+      else
+        from = ((options.dig(:page, :number) || 1) - 1) * (options.dig(:page, :size) || 25)
+        search_after = nil
+      end
+
+      # Calculate cursor sort based upon the type of model
+      # Cursor sorts override any option sort
+      if self.name == "Doi" && !options.dig(:page, :cursor).nil?
         sort = [{ created: "asc", doi: "asc"}]
-      elsif self.name == "Event" && options.dig(:page, :cursor).present?
-        from = 0
-        search_after = [options.dig(:page, :cursor)]
+      elsif self.name == "Event" && !options.dig(:page, :cursor).nil?
         sort = [{ _id: { order: 'asc' }}]
-      elsif self.name == "Activity" && options.dig(:page, :cursor).present?
-        from = 0
-        search_after = [options.dig(:page, :cursor)]
+      elsif self.name == "Activity" && !options.dig(:page, :cursor).nil?
```

Comment:
Yes, because if there is any kind of cursor it should always change the default sort.
Arguable none of this code should be in here and instead just be passed as cursor sort options perhaps, but I think this is an acceptable compromise without refactoring more parts.
Additionally cursor should always be [] not "", I could ensure this with an error check at the start perhaps.

Reply:
you can force an array with `Array.wrap(var)`. And `elsif self.name == "Activity" && options.dig(:page, :cursor)` should be the same as `elsif self.name == "Activity" && !options.dig(:page, :cursor)nil?`